miso-web/src/main/webapp/scripts/sample_hot.js

```diff
@@ -1405,7 +1406,83 @@ Sample.hot = {
     }
     return (Hot.messages.failed.length ? false : true);
   },
+
+  /**
+   * Checks whether the provided sample class alias 
+   */
+  canSampleClassPropagateLibraries: function (sampleClassAlias) {
+    var libraryable = Hot.sampleOptions.sampleClassesDtos.filter(function (sc) { return sc.sampleCategory == 'Aliquot' && sc.alias != 'whole RNA (aliquot)'; });
+    var canPropagate = libraryable.filter(function (sc) { return sc.alias == sampleClassAlias; });
+    return canPropagate.size == 1;
+  },
+  
+  /**
+   * Adds buttons: "Update Samples", "Propagate Samples", "Propagate Libraries" (if appropriate)
+   */
+  getBulkActions: function () {
+    var actions = [];
+    actions.push('<a onclick="Sample.hot.bulkUpdate();" href="javascript:void(0);">Update selected</a>');
+    if (Hot.detailedSample) {
+      Sample.sampleClasses = Hot.sampleOptions.sampleClassesDtos;
+      Sample.validRelationships = Hot.sampleOptions.sampleValidRelationshipsDtos;
+      var uniqueParentScAliases = Sample.ui.getUniqueValues(
+          Hot.startData.map(function (sam) { return sam.sampleClassAlias; })
+      );
+      var uniqueParentSCs = Sample.sampleClasses.filter(function (sc) { return uniqueParentScAliases.indexOf(sc.alias) != -1; });
+      var validChildren = Sample.ui.getChildSampleClasses(uniqueParentSCs);
+      if (validChildren.length > 0) {
+        actions.push('<a style="cursor:pointer" href="javascript:void(0);">Propagate samples:</a>');
```

Comment:
Given we have `Propagate libraries` why not have these all as `Propagate to X`? Does it look terrible?